### ISSUE TITLE
Black screen with underscore after resuming arch linux from a long suspend

### ISSUE THREAD

**Issue by jakeyhanna** [hyprwm/Hyprland#6289]
### System Info and Version
```
<details> <summary>

Hyprland, built from branch main at commit a60c7283e677ed81a466d1c0c864c8a05e192cac  (xwayland: verify new xsurf is valid in prop reads).
Date: Wed May 29 10:34:25 2024
Tag: v0.40.0-144-ga60c7283, commits: 4750

flags: (if any)


System Information:
System name: Linux
Node name: Jarch
Release: 6.9.2-arch1-1
Version: #1 SMP PREEMPT_DYNAMIC Sun, 26 May 2024 01:30:29 +0000


GPU information: 
06:00.0 VGA compatible controller [0300]: NVIDIA Corporation GA106 [GeForce RTX 3060 Lite Hash Rate] [10de:2504] (rev a1) (prog-if 00 [VGA controller])


os-release: NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo


plugins:
```
```
======Config-Start======


Config File: /home/Jack/.config/hypr/hyprland.conf: Read Succeeded

# #######################################################################################
# AUTOGENERATED HYPR CONFIG.
# PLEASE USE THE CONFIG PROVIDED IN THE GIT REPO /examples/hypr.conf AND EDIT IT,
# OR EDIT THIS ONE ACCORDING TO THE WIKI INSTRUCTIONS.
# #######################################################################################

#
# Please note not all available settings / options are set here.
# For a full list, see the wiki
#

autogenerated = 0 # remove this line to remove the warning

# See https://wiki.hyprland.org/Configuring/Monitors/
#monitor=,preferred,auto,auto
#monitor=,highrr,auto,1
monitor=HDMI-A-1,1920x1080@75,0x0,1



# See https://wiki.hyprland.org/Configuring/Keywords/ for more

# Execute your favorite apps at launch
# exec-once = waybar & hyprpaper & firefox

# Source a file (multi-file configs)
# source = ~/.config/hypr/myColors.conf

# Set programs that you use
$terminal = kitty
$fileManager = thunar
$menu = wofi --show drun

# Some default env vars.
#env = XCURSOR_SIZE,24
#env = QT_QPA_PLATFORMTHEME,qt5ct # change to qt6ct if you have that
#env = LIBVA_DRIVER_NAME,nvidia
#env = XDG_SESSION_TYPE,wayland
#env = GBM_BACKEND,nvidia-drm
#env = __GLX_VENDOR_LIBRARY_NAME,nvidia
#env = WLR_NO_HARDWARE_CURSORS,1

env = LIBVA_DRIVER_NAME,nvidia
env = XDG_SESSION_TYPE,wayland
env = GBM_BACKEND,nvidia-drm
env = __GLX_VENDOR_LIBRARY_NAME,nvidia
env = NVD_BACKEND,direct


# For all categories, see https://wiki.hyprland.org/Configuring/Variables/
input {
    kb_layout = us,ara
    #kb_variant = ,qwerty
    #kb_model =
    kb_options = grp:alt_shift_toggle
    #kb_rules =

    follow_mouse = 1

    touchpad {
        natural_scroll = no
    }

    sensitivity = 0 # -1.0 to 1.0, 0 means no modification.
}

general {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more

    gaps_in = 5
    gaps_out = 20
    border_size = 2
    col.active_border = rgba(c53ddcaa) rgba(ff00d666) 45deg
    col.inactive_border = rgba(595959aa)

    layout = master

    # Please see https://wiki.hyprland.org/Configuring/Tearing/ before you turn this on
    allow_tearing = false
}

decoration {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more

    rounding = 10
    
    blur {
        enabled = true
        size = 3
        passes = 1
    }

    drop_shadow = yes
    shadow_range = 4
    shadow_render_power = 3
    col.shadow = rgba(1a1a1aee)
}

animations {
    enabled = yes

    # Some default animations, see https://wiki.hyprland.org/Configuring/Animations/ for more

    bezier = myBezier, 0.05, 0.9, 0.1, 1.05

    animation = windows, 1, 7, myBezier,popin
    animation = windowsOut, 1, 7, default, popin 80%
    animation = border, 1, 10, default
    animation = borderangle, 1, 8, default
    animation = fade, 1, 7, default
    animation = workspaces, 1, 6, default, slidefade
}

dwindle {
    # See https://wiki.hyprland.org/Configuring/Dwindle-Layout/ for more
    pseudotile = yes # master switch for pseudotiling. Enabling is bound to mainMod + P in the keybinds section below
    preserve_split = yes # you probably want this
}

master {
    # See https://wiki.hyprland.org/Configuring/Master-Layout/ for more
    new_is_master = false
}

gestures {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more
    workspace_swipe = off
}

misc {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more
    force_default_wallpaper = -1 # Set to 0 or 1 to disable the anime mascot wallpapers
}

# Example per-device config
# See https://wiki.hyprland.org/Configuring/Keywords/#per-device-input-configs for more
device {
    name = epic-mouse-v1
    sensitivity = -0.5
}

# Example windowrule v1
# windowrule = float, ^(kitty)$
# Example windowrule v2
# windowrulev2 = float,class:^(kitty)$,title:^(kitty)$
# See https://wiki.hyprland.org/Configuring/Window-Rules/ for more
windowrulev2 = suppressevent maximize, class:.* # You'll probably like this.


# See https://wiki.hyprland.org/Configuring/Keywords/ for more
$mainMod = SUPER

# Example binds, see https://wiki.hyprland.org/Configuring/Binds/ for more
bind = $mainMod, Q, exec, $terminal
bind = $mainMod, C, killactive, 
bind = $mainMod, O, exit,
bind = $mainMod, E, exec, $fileManager
bind = $mainMod, V, togglefloating, 
bind = $mainMod, W, exec, $menu
bind = $mainMod, P, pseudo, # dwindle
bind = $mainMod, J, togglesplit, # dwindle
bind = $mainMod, B, exec, firefox

# Move focus with mainMod + arrow keys
bind = $mainMod, left, movefocus, l
bind = $mainMod, right, movefocus, r
bind = $mainMod, up, movefocus, u
bind = $mainMod, down, movefocus, d

# Switch workspaces with mainMod + [0-9]
bind = $mainMod, 1, workspace, 1
bind = $mainMod, 2, workspace, 2
bind = $mainMod, 3, workspace, 3
bind = $mainMod, 4, workspace, 4
bind = $mainMod, 5, workspace, 5
bind = $mainMod, 6, workspace, 6
bind = $mainMod, 7, workspace, 7
bind = $mainMod, 8, workspace, 8
bind = $mainMod, 9, workspace, 9
bind = $mainMod, 0, workspace, 10

# Move active window to a workspace with mainMod + SHIFT + [0-9]
bind = $mainMod SHIFT, 1, movetoworkspace, 1
bind = $mainMod SHIFT, 2, movetoworkspace, 2
bind = $mainMod SHIFT, 3, movetoworkspace, 3
bind = $mainMod SHIFT, 4, movetoworkspace, 4
bind = $mainMod SHIFT, 5, movetoworkspace, 5
bind = $mainMod SHIFT, 6, movetoworkspace, 6
bind = $mainMod SHIFT, 7, movetoworkspace, 7
bind = $mainMod SHIFT, 8, movetoworkspace, 8
bind = $mainMod SHIFT, 9, movetoworkspace, 9
bind = $mainMod SHIFT, 0, movetoworkspace, 10

# Example special workspace (scratchpad)
bind = $mainMod, S, togglespecialworkspace, magic
bind = $mainMod SHIFT, S, movetoworkspace, special:magic

# Scroll through existing workspaces with mainMod + scroll
bind = $mainMod, mouse_down, workspace, e+1
bind = $mainMod, mouse_up, workspace, e-1

# Move/resize windows with mainMod + LMB/RMB and dragging
bindm = $mainMod, mouse:272, movewindow
bindm = $mainMod, mouse:273, resizewindow

#bind=, Print, exec, grimshot copy area

bind=, Print, exec, bash /run/media/Games2/programming/scripts/flameshot2.sh

bind=$mainMod,M,fullscreen,1
bind=$mainMod,F,fullscreen,

bind = SUPER ALT, H, movewindow, l
bind = SUPER ALT, L, movewindow, r
bind = SUPER ALT, K, movewindow, u
bind = SUPER ALT, J, movewindow, d
bindel=, XF86AudioRaiseVolume, exec, wpctl set-volume @DEFAULT_AUDIO_SINK@ 5%+
bindel=, XF86AudioLowerVolume, exec, wpctl set-volume @DEFAULT_AUDIO_SINK@ 5%-
bindl=, XF86AudioMute, exec, wpctl set-mute @DEFAULT_AUDIO_SINK@ toggle



bind = ALT,F10,pass,^(com\.obsproject\.Studio)$
#bind = CTRL,T,pass,firefox

exec-once=waybar
exec-once=dunst
exec-once=hypridle
exec-once=hyprpaper
exec-once='/usr/lib/polkit-kde-authentication-agent-1'
exec-once='systemctl --user enable opentabletdriver'

#exec-once=swaybg --i /home/Jack/Pictures/'Evil fucking cool skeelton.png'

======Config-End========</summary>
```
</details>

### Bug or Regression?

Bug

### Description

I leave my Arch for a long time (An hour to 2 hours), and when i resume it with my keyboard it gives me a black screen with an underscore on the top left blinking (Important) before sending me back to sddm.

I log in again to hyprland with everything that i had opened being closed/Fresh start and also with network manager needing to be restarted due to the internet not working.

I restart it using `systemctl restart NetworkManager.service`.

Here is the real problem, When i press crtl-c the system just crashes and it sends me back to the black screen with the underscore with this time the underscore not blinking, I can't acess any tty after that happened and i am forced to do a hard restart.

### How to reproduce

Unknown

Started to happen suddenly out of the blue

### Crash reports, logs, images, videos

[hyprlandCrashReport166.txt](https://github.com/user-attachments/files/15516807/hyprlandCrashReport166.txt)
[hyprlandCrashReport855.txt](https://github.com/user-attachments/files/15516808/hyprlandCrashReport855.txt)
[hyprlandCrashReport894.txt](https://github.com/user-attachments/files/15516809/hyprlandCrashReport894.txt)
[hyprlandCrashReport1766.txt](https://github.com/user-attachments/files/15516810/hyprlandCrashReport1766.txt)
![image](https://github.com/hyprwm/Hyprland/assets/49160062/6c43d677-f889-47a0-a4a1-b079c992c495)

**Comment by jakeyhanna** [comment]
This is my first bug report ever, Please tell me if anyone needs more info

**Comment by Aqa-Ib** [comment]
>  When i press crtl-c the system just crashes

You press ctrl-c where?

**Comment by jakeyhanna** [comment]
Sorry for the confusion, When i log back into hyprland after returning to sddm and i press ctrl-c anywhere while in hyprland (Doesn't matter if it's around text or not) the system does the crash

**Comment by jakeyhanna** [comment]
Thinking about it, isn't ctrl-c how you stop a process in the terminal?, What if that has something to do with the bug?

**Comment by Aqa-Ib** [comment]
Seems like some problem with the nvidia drivers, look at this in your logs: ```[CRITICAL] m_sWLRBackend was NULL! This usually means wlroots could not find a GPU or enountered some issues.```

I would try removing these environment variables from your hyprland.conf:
```
env = LIBVA_DRIVER_NAME,nvidia
env = XDG_SESSION_TYPE,wayland
env = GBM_BACKEND,nvidia-drm
env = __GLX_VENDOR_LIBRARY_NAME,nvidia
env = NVD_BACKEND,direct
```
Restart the session and see what happen.

If you are not into gaming, I would definitely ditch the closed source nvidia driver and just install the mesa driver nouveau, as it is far less conflictive.

**Comment by jakeyhanna** [comment]
I am into gaming so sadly i won't switch to the nouveau driver, Although i will try the change you suggested

Does it affect anything if i removed these variables??

**Comment by Aqa-Ib** [comment]
yea, regarding hardware acceleration for watching videos.

**Comment by Aqa-Ib** [comment]
Just try and see if any of those variables is conflicting in your system.

**Comment by jakeyhanna** [comment]
Alright, Thank you for the fast replies and help btw!!

**Comment by jakeyhanna** [comment]
I will try disabling them

**Comment by Aqa-Ib** [comment]
Maybe `env = GBM_BACKEND,nvidia-drm ` is needed, though. 
`env = XDG_SESSION_TYPE,wayland` this one is automatically defined in my system, but you can leave it there just in case.

**Comment by jakeyhanna** [comment]
Alright, I am going to only enable this, If the problem continues i will disable it, If the problem continues after disabling it i will be in contact, Although i am sorry if the reply will be slow

**Comment by Aqa-Ib** [comment]
One thing that I missed to ask you: does the ctrl-c problem happen in a fresh session that did not crash after suspending?

**Comment by jakeyhanna** [comment]
> One thing that I missed to ask you: does the ctrl-c problem happen in a fresh session that did not crash after suspending?

No, it only happens after hyprland has crashed and sent me back to sddm

On a restart or on a fresh session it functions like normal (copy)

**Comment by Aqa-Ib** [comment]
Do you have nvidia-suspend and nvidia-resume services enabled?

**Comment by jakeyhanna** [comment]
> Do you have nvidia-suspend and nvidia-resume services enabled?

Yep, I have enabled them, And they are active when i do systemctl status on them

**Comment by Aqa-Ib** [comment]
Archwiki says that those require a kernel parameter and are still experimental. If you have the kernel parameter active, try disabling it and the services altogether, maybe it can suspend without those. If you dont have the kernel parameter active, maybe that's what is missing. 

https://wiki.archlinux.org/title/NVIDIA/Tips_and_tricks#Preserve_video_memory_after_suspend

**Comment by jakeyhanna** [comment]
Will try!!

**Comment by jakeyhanna** [comment]
Update: I kept `env = GBM_BACKEND,nvidia-drm` And `env = XDG_SESSION_TYPE,wayland`

The problem still presisted, so i am disabling them

As to the nvidia services, When i disabled them my screen would turn off but then return suddenly to the hyprlock screen meaning it failed to suspend

I am going to wait on if disabling the rest of the envs will cause the same effect

There isn't any new crash report in .cache/hyprland

Also sorry for the wait @Aqa-Ib, It takes a while to take effect so i can diagnose it sadly

**Comment by Aqa-Ib** [comment]
Play with those, hopefully you can get it running. 

If you can't though, just wait some time because nvidia is doing a lot of work lately on their linux drivers. Hopefully they fix all the problems they have soon enough.

**Comment by jakeyhanna** [comment]
> Play with those, hopefully you can get it running.
> 
> If you can't though, just wait some time because nvidia is doing a lot of work lately on their linux drivers. Hopefully they fix all the problems they have soon enough.

Alright i will try, speaking on nvidia, The problem started while i was using the 535xx-dkms drivers since they are the best drivers for wayland Previously

I hope time fixes them

**Comment by jakeyhanna** [comment]
Thank you a lot by the way Aqa-Ib for the fast reply!!

**Comment by jakeyhanna** [comment]
Off note: wayland just updated from 1.22 to 1.23, Hope this helps a bit

**Comment by vaxerski** [comment]
this is an nvidia issue, I can't help you much here. Feel free to ask around on discord and search stuff online.

**Comment by jakeyhanna** [comment]
This problem was also reported for a person with an AMD gpu when asked on reddit

**Comment by jakeyhanna** [comment]
And this doesn't happen on KDE wayland, Just hyprland

**Comment by vaxerski** [comment]
I can't find the amd person here, they can open another issue if they are affected

**Comment by jakeyhanna** [comment]
https://www.reddit.com/r/hyprland/comments/1d4sqo2/hyprland_sometimes_gets_me_a_black_screen_with_an/
